antarest/study/repository.py

```diff
@@ -44,34 +66,59 @@ def save(
     def refresh(self, metadata: Study) -> None:
         db.session.refresh(metadata)
 
-    def get(self, id: str) -> Optional[Study]:
+    def get(self, id: str) -> t.Optional[Study]:
         """Get the study by ID or return `None` if not found in database."""
-        metadata: Study = db.session.query(Study).get(id)
+        # When we fetch a study, we also need to fetch the associated owner and groups
+        # to check the permissions of the current user efficiently.
+        metadata: Study = (
+            # fmt: off
+            db.session.query(Study)
+            .options(joinedload(Study.owner))
+            .options(joinedload(Study.groups))
+            .get(id)
+            # fmt: on
+        )
         return metadata
 
     def one(self, id: str) -> Study:
         """Get the study by ID or raise `sqlalchemy.exc.NoResultFound` if not found in database."""
-        study: Study = db.session.query(Study).filter_by(id=id).one()
+        # When we fetch a study, we also need to fetch the associated owner and groups
+        # to check the permissions of the current user efficiently.
+        study: Study = (
+            db.session.query(Study)
+            .options(joinedload(Study.owner))
+            .options(joinedload(Study.groups))
+            .filter_by(id=id)
+            .one()
+        )
         return study
 
-    def get_list(self, study_id: List[str]) -> List[Study]:
-        studies: List[Study] = db.session.query(Study).where(Study.id.in_(study_id)).all()
+    def get_list(self, study_id: t.List[str]) -> t.List[Study]:
+        # When we fetch a study, we also need to fetch the associated owner and groups
+        # to check the permissions of the current user efficiently.
+        studies: t.List[Study] = (
+            db.session.query(Study)
+            .options(joinedload(Study.owner))
+            .options(joinedload(Study.groups))
+            .where(Study.id.in_(study_id))
+            .all()
+        )
         return studies
 
-    def get_additional_data(self, study_id: str) -> Optional[StudyAdditionalData]:
+    def get_additional_data(self, study_id: str) -> t.Optional[StudyAdditionalData]:
         metadata: StudyAdditionalData = db.session.query(StudyAdditionalData).get(study_id)
         return metadata
 
-    def get_all(self) -> List[Study]:
+    def get_all(self) -> t.List[Study]:
         entity = with_polymorphic(Study, "*")
-        metadatas: List[Study] = db.session.query(entity).filter(RawStudy.missing.is_(None)).all()
+        metadatas: t.List[Study] = db.session.query(entity).filter(RawStudy.missing.is_(None)).all()
```

Comment:
`metadatas` as singular: `metadata`
also consider using consistent naming, here `study` `studies` and `metadata` are used, `metadata` looks good in my opinion 
> you can also update it in `get_all_raw` `get_list` etc